### ISSUE TITLE
Ray Serve: Additional Flexibility When Configuring Placement Strategy

### ISSUE THREAD

**Issue by ewellinger** [ray-project/ray#45885]
### Description

This feature request is to allow for setting the `max_replicas_per_node` setting while also directly specifying the placement group bundles (currently setting both is not allowed).

We are configuring our deployments by explicitly specifying the placement group bundles for the actor and worker processes. For the `placement_group_strategy` we are setting it to STRICT_PACK to ensure that all these resources are colocated with each other. For example we configure our deployment with the following bundles:

```python
[{'CPU': 1, 'GPU': 0, 'accelerator_type:A10G': 0.001},
 {'CPU': 8.0, 'GPU': 1.0, 'accelerator_type:A10G': 0.001},
 {'CPU': 8.0, 'GPU': 1.0, 'accelerator_type:A10G': 0.001}]
```

This works well for the placement of the replica's resources, but we also need to ensure that replicas are spread across nodes to guarantee better resiliency which is not currently possible.

### Use case

The [guidance for end-to-end fault tolerance](https://docs.ray.io/en/latest/serve/production-guide/fault-tolerance.html#spreading-replicas-across-nodes) recommends setting `max_replicas_per_node` to ensure that a node failure won't take out all available replicas. Leveraging that option isn't currently possible if you also need to specify the placement group bundles directly.

**Comment by ewellinger** [comment]
And if it helps with prioritization, this has lead to outages where a deployments only replicas were colocated on a node which went down. Our short-term "fix" is to ensure at least 3 replicas are running which will force it to be spread between two nodes due to GPU constraints.

**Comment by anyscalesam** [comment]
@jjyao does our planned work for label based scheduling enable this; may be coupled with a scheduling strategy that is implemented at the serve layer (but core needs to provide a way to do this for serve).

**Comment by jjyao** [comment]
Hi @ewellinger for STRICT_PACK PG, we can support max_replicas_per_node with a clear semantics. But for non-strict pack PG, the meaning of max_replicas_per_node is unclear.

**Comment by jjyao** [comment]
@zcin we can first support it with STRICT_PACK PG only.

**Comment by ewellinger** [comment]
Yup, that makes sense.  For our purposes we're only setting `STRICT_PACK` for these placement groups but I think it's a reasonable restriction that you can't both set `max_replicas_per_node` and specify a non-strict packed placement group.

**Comment by pksukumar** [comment]
It would be great, If we can specify lifetime of the placement group using below annotation,

@serve.deployment(
    name="FlagEmbeddings",
    placement_group_strategy="SPREAD",
    placement_group_bundles=[{"CPU": 1,"GPU":0.3},{"GPU":0.3}],
    **placement_group_lifetime="detached"**
)
Currently this is not possible.
